sdk/provisioning/Azure.Provisioning.PostgreSql/src/PostgreSqlFlexibleServerConfiguration.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+using Azure.ResourceManager.PostgreSql.FlexibleServers;
+using Azure.ResourceManager.PostgreSql.FlexibleServers.Models;
+
+namespace Azure.Provisioning.PostgreSql
+{
+    /// <summary>
+    /// Represents a PostgreSql flexible server configuration.
+    /// </summary>
+    public class PostgreSqlFlexibleServerConfiguration : Resource<PostgreSqlFlexibleServerConfigurationData>
+    {
+        private const string ResourceTypeName = "Microsoft.DBforPostgreSQL/flexibleServers/configurations";
+        private static PostgreSqlFlexibleServerConfigurationData Empty(string name)
+            => ArmPostgreSqlFlexibleServersModelFactory.PostgreSqlFlexibleServerConfigurationData();
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="PostgreSqlFlexibleServerConfiguration"/> class.
+        /// </summary>
+        /// <param name="scope">The scope.</param>
+        /// <param name="value">The value.</param>
+        /// <param name="source">The source.</param>
+        /// <param name="configPropertyName">The config property name.</param>
+        /// <param name="parent">The parent.</param>
+        /// <param name="name">The name.</param>
+        /// <param name="version">The version.</param>
+        public PostgreSqlFlexibleServerConfiguration(
+            IConstruct scope,
+            string value,
+            string source = "user-override",
+            string configPropertyName = "azure.extensions",
+            PostgreSqlFlexibleServer? parent = null,
+            string name = "config",
+            string version = PostgreSqlFlexibleServer.DefaultVersion)
```

Comment:
What do you think about a signature more like maybe

```csharp
public PostgreSqlFlexibleServerConfiguration(
    IConstruct scope,
    string propertyName,
    string propertyValue,
    string propertySource = "user-override",
    PostgreSqlFlexibleServer? parent = null,
    string name = "config",
    string version = PostgreSqlFlexibleServer.DefaultVersion)
```
to make it easier to map the flattened properties into bicep like

```bicep
resource foo 'Microsoft.DBforPostgreSQL/flexibleServers/configurations@2023-03-01-preview' = {
  parent: bar
  name: '<<propertyName>>'
  properties: {
    value: '<<propertyValue>>'
    source: 'user-override'
  }
}
```

Reply:
Yep, that makes sense, I'll refactor it. And it probably is a good idea to remove the default value from `propertyName` since I'm not sure what the options that can go there are beyond `azure.extensions`, but I'd assume there are more.

Reply:
Refactored to align with suggestion